fsm.go

```diff
@@ -87,6 +91,15 @@ type Events []EventDesc
 // Callbacks is a shorthand for defining the callbacks in NewFSM.
 type Callbacks map[string]Callback
 
+// DataValue is stored in data which can be indexed using key. It has Value to
+// store underlying data and ValueMu used for safely storing and retrieving
+type DataValue struct {
```

Comment:
Why is this needed? I think the map can be `map[string]interface{}` directly?

Reply:
I have done that so that locking can be fine grained (key level locking). We can just use `map[string]interface{}` with RWMutex if it seems to be an overkill.

Reply:
Yes, I think so. There could be a reference stored inside anyway, so not worth protecting individual values.